github/issues_test.go

```diff
@@ -240,3 +240,29 @@ func TestIssuesService_Edit_invalidOwner(t *testing.T) {
 	_, _, err := client.Issues.Edit("%", "r", 1, nil)
 	testURLParseError(t, err)
 }
+
+func TestIssuesService_Lock(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/issues/1/lock", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PUT")
```

Comment:
to better simulate the actual GitHub API, go ahead and have this return a 204 status:

```go
w.WriteHeader(http.StatusNoContent)
```

same with Unlock test below